app/models/github_status.rb

```diff
@@ -48,15 +48,15 @@ def self.fetch(repo, ref)
     # Fall back to a "missing" status.
     return new("missing", []) if response.nil?
 
-    # Don't cache pending statuses, since we expect updates soon.
-    unless response.state == "pending"
-      Rails.cache.write(cache_key, response, expires_in: 1.hour)
-    end
```

Comment:
The state is `failed` if any of the checks have failed, even if others are pending.